packages/node-http-handler/src/set-socket-keep-alive.ts

```diff
@@ -0,0 +1,18 @@
+import { ClientRequest } from "http";
+
+const DEFAULT_INITIAL_DELAY = 10 * 1000;
+
+export interface SocketKeepAliveOptions {
+    keepAlive: boolean;
+    initialDelay: number;
+}
+
+export const setSocketKeepAlive = (request: ClientRequest, socketKeepAlive: SocketKeepAliveOptions) => {
+    if (!socketKeepAlive || socketKeepAlive.keepAlive === false) {
+        return;
+    }
+
+    request.on('socket', (socket) => {
+        socket.setKeepAlive(socketKeepAlive.keepAlive, socketKeepAlive.initialDelay || DEFAULT_INITIAL_DELAY);
```

Comment:
Looks like `keepAlive` always needs to be true for `initialDelay` to be effective.

Should we simplify option `socketKeepAlive: SocketKeepAliveOptions` to `socketKeepAliveInitialDelay: number`?

Reply:
What I did was to make the initialDelay parameter optional, so that if is not provided then it falls into the default value.